src/nix/dev-shell.cc

```diff
@@ -281,6 +284,40 @@ struct CmdDevShell : Common, MixEnvironment
         };
     }
 
+    std::string getBashPath(ref<Store> store)
```

Comment:
Maybe this function can be generalized to fetch any package. For example, it could be used by the Git fetcher to fetch `nixpkgs#git` it `git` is not installed. Although that's a bit tricky right now because libfetcher doesn't have access to libexpr.

Reply:
Could I start by taking out the SourceExprCommand from InstallableValue? Looks like all it's used for is state, getAutoArgs, and lockFlags, and it might be simpler for an Installable to be separate from a command. Then Installables could be used by InstallableCommands/fetchGit/dev-shell.